src/lib-components/forms/DateTimeInput.vue

```diff
@@ -6,36 +6,27 @@ import {
   useInputField,
   defaultInputProps,
   toDatetimeLocal,
+  defaultModelOpts,
 } from "@/composables/forms"
 import type { DateTimeInput } from "@/composables/forms"
-import { computed, ref } from "vue"
 
 defineOptions({
   inheritAttrs: false,
 })
 
 const props = withDefaults(defineProps<DateTimeInput>(), defaultInputProps)
+const modelState = defineModel<DateTimeInput["modelValue"]>({
+  ...defaultModelOpts,
+  get: (v) => {
+    return v ? toDatetimeLocal(v) : v
+  },
+  set: (v) => {
+    return v ? new Date(v).toISOString() : v
+  },
+})
```

Comment:
This was one of the nicest instances where the `defineModel` features really cleaned things up.